### PR TITLE
Accuracy per tag optimization

### PR DESCRIPTION
**Pull request by danielhertenstein**
Addresses #104 

Previously, `accuracy_per_tag()` ran each sample through the model one at a time and then checked the accuracy for that sample. This pull request runs all the samples through the model at the same time and then computes the accuracy metrics for all the samples using numpy methods.

There is an additional change to `accuracy_per_tag()` such that it now takes predictions as an argument instead of the model and the samples. This is because the main source of calls for `accuracy_per_tag()` is `learn()` which itself runs all the samples through the model and therefore has the predictions laying around for use. This change requires the last calls to `accuracy_per_tag()`, located within `main()`, to run the samples through the model beforehand.

The new overall profile shows that the overall runtime and the runtime for `accuracy_per_tag()` are much lower.
new_overall_profile

### CODE REVIEW

File: cli/fathom_web/train.py [hunk 1]
```diff
@@ -290,10 +294,12 @@ def main(training_file, validation_file, stop_early, learning_rate, iterations,
         validation_arg = None
     model = learn(learning_rate, iterations, x, y, validation=validation_arg, stop_early=stop_early, run_comment=full_comment)
     print(pretty_coeffs(model, training_data['header']['featureNames']))
-    accuracy, false_positive, false_negative = accuracy_per_tag(model, x, y)
+    y_pred = model(x)  # Make predictions.
```
Comment (erikrose) [inline cli/fathom_web/train.py]: I actually slightly prefer inlining `model(x)` into the following function call (and the same below, for validation). I don't think it's any harder to comprehend (a look at the name of the param it's getting passed into in accuracy_per_tag() clues you in if not), and it doesn't add more state to an already pretty stateful procedure. What do you think?

File: cli/fathom_web/train.py [hunk 2]
```diff
@@ -82,18 +83,21 @@ def learn(learning_rate, iterations, x, y, validation=None, stop_early=False, ru
     return model
 
 
-def accuracy_per_tag(model, x, y):
+def accuracy_per_tag(y, y_pred):
     """Return the accuracy 0..1 of the model on a per-tag basis, given input
```
Comment (erikrose) [inline cli/fathom_web/train.py]: Docstring needs an update since the params changed.

File: cli/fathom_web/train.py (hunk 1 above)
Comment (danielhertenstein) [reply cli/fathom_web/train.py]: I'm on board with this.

File: cli/fathom_web/train.py (hunk 2 above)
Comment (danielhertenstein) [reply cli/fathom_web/train.py]: Whoops. I will update that.
